flink/v1.16/flink/src/test/java/org/apache/iceberg/flink/source/TestIcebergSourceContinuous.java

```diff
@@ -325,7 +326,7 @@ public void testSpecificSnapshotTimestamp() throws Exception {
     long snapshot0Timestamp = tableResource.table().currentSnapshot().timestampMillis();
 
     // sleep for 2 ms to make sure snapshot1 has a higher timestamp value
-    Thread.sleep(2);
+    Awaitility.await().pollDelay(2, TimeUnit.MILLISECONDS).until(() -> System.currentTimeMillis()-snapshot0Timestamp>2);
```

Comment:
IMHO, let's not use `Awaitility` for just system clock delay. We can enhance [TestHelpers](https://github.com/apache/iceberg/blob/dd02085b12e9b509efef2d9a8b9fe730c0c60b03/api/src/test/java/org/apache/iceberg/TestHelpers.java#L57)  like
`public static long waitUntil1ms() {
long previous = System.currentTimeMillis();
    long current = System.currentTimeMillis();
    while (current <= previous) {
      current = System.currentTimeMillis();
    }
    return current;
  }`
for any other `waitUntilAfter` we can use  `Awaitility`